src/ICSharpCode.SharpZipLib/Tar/TarHeader.cs

```diff
@@ -870,9 +870,11 @@ public static int GetNameBytes(string name, int nameOffset, byte[] buffer, int b
 
 			int i;
 
-			for (i = 0; i < length && nameOffset + i < name.Length; ++i)
+			byte[] nameBytes = Encoding.UTF8.GetBytes(name);
```

Comment:
I doubt we want BOMs in the name

```suggestion
			byte[] nameBytes = new Encoding.UTF8Encoding(false).GetBytes(name);
```